src/bindings/g_object/type.cr

```diff
@@ -98,6 +98,19 @@ module GObject
   end
 end
 
+class Object
+  # Returns the GObject GType for pure Crystal reference objects.
+  def self.g_type
+    GObject::TYPE_POINTER
+  end
+
+  def to_unsafe
```

Comment:
This method can (and will) be overridden.
This is a very unsafe way of doing things and sooner or later this will cause a memory corruption issue on this repo by some user trying to pass (for example) a `XML::Node` via a signal.

Reply:
I’m not happy with this either, if someone forget to define `to_unsafe` for something the code will compile and send a bad pointer to the C function.